packages/js/components/src/split-dropdown/split-dropdown.tsx

```diff
@@ -0,0 +1,68 @@
+/**
+ * External dependencies
+ */
+import { cloneElement, createElement } from '@wordpress/element';
+import { Icon, chevronDown, chevronUp } from '@wordpress/icons';
+import { Button, ButtonGroup, Dropdown } from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+export type SplitDropdownProps = {
+	className?: string;
+	disabled?: boolean;
+	variant?: string;
+	children: JSX.Element[];
+};
+
+export const SplitDropdown: React.FC< SplitDropdownProps > = ( {
+	className = '',
+	children,
+	disabled,
+	variant = 'primary',
+}: SplitDropdownProps ) => {
+	const groupActionProps = Object.assign(
+		{ variant },
+		disabled ? { disabled } : {}
+	);
+	const mainActionProps = {
+		...groupActionProps,
+		className: `woocommerce-split-dropdown__main-button ${ className }`,
+	};
+	const [ mainAction, ...menuActions ] = children;
+	const menuIcon = chevronDown;
+	const menuIconExpanded = chevronUp;
+	return (
+		<ButtonGroup className={ `woocommerce-split-dropdown ${ className }` }>
+			{ cloneElement( mainAction, mainActionProps ) }
+			<Dropdown
```

Comment:
@louwie17 made a comment about using the [DropdownMenu](https://wordpress.github.io/gutenberg/?path=/story/components-dropdownmenu--default&knob-Body%20Title=My%20Block%20Settings&knob-Card:%20elevation_Card=0&knob-Card:%20isRounded_Card=true&knob-Card:%20size_Card=large&knob-Card:%20width%20(via%20inline%20styles)_Card=360&knob-CardBody:%20size_CardBody=unset&knob-CardFooter:%20isBorderless_CardFooter=unset&knob-CardFooter:%20justify_CardFooter=space-between&knob-CardFooter:%20size_CardFooter=unset&knob-CardHeader:%20isBorderless_CardHeader=unset&knob-CardHeader:%20size_CardHeader=unset&knob-Is%20Dismissible=true&knob-Label=Default%20Button&knob-Opened=true&knob-Row%20Text=My%20Panel%20Inputs%20and%20Labels&knob-Status=info&knob-focusOnMount=true&knob-isDismissible=true&knob-shouldCloseOnClickOutside=true&knob-shouldCloseOnEsc=true&knob-title=Title) component.  I'm not opposed to this as it simplifies the menu, but obviously comes with some limitations in what can be shown in the dropdown.

I guess the question is whether or not we see ourselves using anything besides default buttons in the dropdown in the future.

Reply:
Unless I'm missing something about `DropdownMenu`, it would need to be modified to let the icon change when the menu is expanded if we're going to match the design.